Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -49,6 +49,8 @@ class TabLocationView: UIView, FeatureFlaggable {
             updateTextWithURL()
             trackingProtectionButton.isHidden = !isValidHttpUrlProtocol
             shareButton.isHidden = !(shouldEnableShareButtonFeature && isValidHttpUrlProtocol)
+            let product = url.flatMap(ShoppingProduct.init)
+            let isShoppingCartButtonVisible = product?.isShoppingCartButtonVisible ?? false
```

Comment:
```suggestion

            var isShoppingCartButtonVisible = false
            if let url = url {
                let product = ShoppingProduct(url: url)
                isShoppingCartButtonVisible = product.isShoppingCartButtonVisible
            }
            
```

Reply:
Oh, do we really want that? It's so much code that's handled specifically by the ternary operator.